src/Reactor.cc

```diff
@@ -734,6 +750,19 @@ bool Reactor::signalEvent(const Event& event)
 {
 	std::visit(overloaded{
 		[&](const QuitEvent& /*e*/) {
+			// check whether we should store the current setup
+			if (auto* board = getMotherBoard()) {
+				auto depth = saveSetupAtExitDepthSetting->getEnum();
+				auto name = saveSetupAtExitNameSetting->getString();
+				if (depth != MSXMotherBoard::SetupDepth::NONE && !name.empty()) {
+					auto filename = FileOperations::parseCommandFileArgument(
+						name, Reactor::SETUP_DIR, "",
+						Reactor::SETUP_EXTENSION);
+
+					board->storeAsSetup(filename, depth);
```

Comment:
I did an experiment:
* Enable save-setup-on-exit, then exit openMSX, this creates `last_used.oms`.
* Then `chmod 400 last_used.oms`. IOW make this file write-protected.
* Start and exit openMSX.

But then on exit, openMSX crashes with:
```
Uncaught exception: could not write "/home/wouter/.openMSX/setups/last_used.oms"
```
So I think here we should catch the exception thrown from `storeAsSetup()` and emit it as a warning. Though I'm not sure emitting a warning is sufficient, because we're about to exit openMSX. And then, depending on how openMSX is configured(?), that warning may not be visible. Maybe we should also print it to `std::cerr`???